src/js/fs.js

```diff
@@ -264,16 +264,26 @@ fs.readFileSync = function(path) {
 };
 
 
-fs.writeFile = function(path, data, callback) {
+fs.writeFile = function(path, data, options, callback) {
+  if (util.isFunction(options)) {
+    callback = options;
+  }
   checkArgString(path);
   checkArgBuffer(data);
   checkArgFunction(callback);
 
   var fd;
   var len;
   var bytesWritten;
+  var flag = 'w';
+  if (util.isObject(options) && !util.isNullOrUndefined(options.flag)) {
+    var opt = options.flag;
+    if (opt === 'a' || opt === 'w') {
```

Comment:
`flag` can be anything other than `a`and `w`. I don't think if statement is needed.